apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -656,6 +656,11 @@ type ClusterParameters struct {
 	// +kubebuilder:validation:Minimum=1
 	// +optional
 	PerConnectionBufferLimitBytes *uint32 `json:"per-connection-buffer-limit-bytes,omitempty"`
+
+	// UpstreamTLS contains the TLS policy parameters for upstream connections
+	//
+	// +optional
+	UpstreamTLS *EnvoyTLS `json:"upstream-tls,omitempty"`
```

Comment:
Question: What do people prefer: nesting TLS config in `ContourConfiguration.spec.envoy.cluster.upstream-tls` (like suggested here) or bumping it one level `ContourConfiguration.spec.envoy.upstream-tls`? 

Maybe bit unfortunate but there is existing upstream TLS parameter directly under `ContourConfiguration.spec.envoy`, which is `clientCertificate`. It's bit unfortunate that the upstream TLS settings will be split in two places, but hopefully not too far from each other.

The corresponding downstream TLS setting is `ContourConfiguration.spec.envoy.listener.tls`. I guess perfect symmetry across TLS settings is not achievable anymore, not even in CRD config which was the fresh start :)

Reply:
I could move clientCertificate into UpstreamTLS. OR would you prefer that be in a separate MR?

Reply:
Let's hold off for now to see what others think about that change, we can do it in a separate PR if we decide to proceed.

Reply:
+1 for keeping it under cluster